ocs_ci/ocs/resources/cloud_manager.py

```diff
@@ -0,0 +1,225 @@
+import base64
+import logging
+import os
+from abc import ABC, abstractmethod
+import boto3
+from botocore.exceptions import ClientError
+from time import sleep
+from google.cloud import storage
+from google.auth.exceptions import DefaultCredentialsError
+import ibm_boto3
+
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants
+from ocs_ci.utility import templating
+from tests.helpers import create_resource
+
+logger = logging.getLogger(name=__file__)
+
+
+class CloudManager(ABC):
+    aws_client, google_client, azure_client, s3comp_client = (None,) * 4
+
+    def __init__(self):
+        # TODO: solve credentials for clients
+        self.aws_client = S3Client()
+        self.google_client = GoogleClient()
+        self.azure_client = AzureClient()
+        self.s3comp_client = S3Client()
+
+
+class CloudClient(ABC):
+    """
+        Base abstract class for Cloud based Underlying Storage
+    """
+    client = None
+
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def create_uls(self, name):
+        """
```

Comment:
text is not aligned
Please fix in all relevant places

Reply:
fixed